### ISSUE TITLE
Global variable cause crash and load gdextension failed

### ISSUE THREAD

**Issue by pupil1337** [godotengine/godot-cpp#1449]
### Godot version

4.3.dev(11d3768)

### godot-cpp version

4.3.dev(e23b117)

### System information

win11

### Issue description

This issue have existed for a long time, The reproduction steps are the same as this comment: https://github.com/godotengine/godot-cpp/issues/358#issuecomment-566168170.
This issue is quite tricky, as it is caused by godot **load gdextension .dll** and **initializate gdextension internal:: function point**
order.
Do need to solve this problem?
(This may be solved by Godot, but I haven't used any other gdextension language, so the report is here.)

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

**Comment by dsnopek** [comment]
Well, we can't make Godot types work if they are constructed before the GDExtension has been initialized.

It would be nice to not crash, although, I'm not sure how feasible that is. We could wrap everything in a check to see if it's been initialized yet, and then output a message, I guess?

**Comment by pupil1337** [comment]
There maybe two types that need to be checked: 
1. function pointer type define in `gdextension_interface.h` (used in godot.cpp)
2. `_method_bindings` in the Variant type(for example in string_name.cpp).

Checking if they are empty requires writing a big logic and the final outcome still cannot declare global variables. I think it's better to not solved it (global variable not many usage scenarios and it's not necessary).

**Comment by OffsetMOSFET** [comment]
Doing some experimentation, one type of global storage does work: static variables inside of functions.

```C++
int Foo::get_increment()
{
  static int counter {0};
  ++counter;
  return counter;
}
```

**Comment by dsnopek** [comment]
> one type of static storage does work: static variables inside of functions

Yes, if the function is called after the GDExtension has been initialized, that should work.

**Comment by dsnopek** [comment]
> I think we can initialize the gdextension-interface function pointer as a `FuncA` instead of nullptr, `FuncA` output an error: "Call constructor too early, earlier than GDExtendenInterface initialization, please check if global godot variables are declared" (similar error output and crash).

Ooo, that's an interesting idea! However, it would be nice to say in the error message the name of the object that's being constructed, and I think the only way to do that would be to have a `FuncA` for each class. I think that's doable, though.
